Block/BrowseAbandonment.php

```diff
@@ -0,0 +1,293 @@
+     */
+    const ELIGIBLE_MAGENTO_PAGE_TYPES_FOR_BROWSE_ABANDONMENT = [
+        self::MAGENTO_HOME_PAGE_TYPE,
+        self::MAGENTO_CATEGORY_PAGE_TYPE,
+        self::MAGENTO_PRODUCT_PAGE_TYPE,
+        self::MAGENTO_ORDER_CREATED_PAGE_TYPE
+    ];
+
+    /**
+     * @const string
+     */
+    const BROWSE_ABANDONMENT_HOME_PAGE_TYPE = 'home';
+
+    /**
+     * @const string
+     */
+    const BROWSE_ABANDONMENT_COLLECTION_PAGE_TYPE = 'collection';
+
+    /**
+     * @const string
+     */
+    const BROWSE_ABANDONMENT_PRODUCT_PAGE_TYPE = 'product';
+
+    /**
+     * @const string
+     */
+    const BROWSE_ABANDONMENT_ORDER_CREATED_PAGE_TYPE = 'order_created';
+
+    /**
+     * @const array<string, string>
+     */
+    const MAGENTO_PAGE_TYPE_TO_BROWSE_ABANDONMENT_PAGE_TYPE_MAP = [
+        self::MAGENTO_HOME_PAGE_TYPE => self::BROWSE_ABANDONMENT_HOME_PAGE_TYPE,
+        self::MAGENTO_CATEGORY_PAGE_TYPE => self::BROWSE_ABANDONMENT_COLLECTION_PAGE_TYPE,
+        self::MAGENTO_PRODUCT_PAGE_TYPE => self::BROWSE_ABANDONMENT_PRODUCT_PAGE_TYPE,
+        self::MAGENTO_ORDER_CREATED_PAGE_TYPE => self::BROWSE_ABANDONMENT_ORDER_CREATED_PAGE_TYPE
+    ];
+
+    /**
+     * @const array<string>
+     */
+    const BROWSE_ABANDONMENT_PAGE_TYPES_ELIGIBLE_FOR_ID = [
+        self::BROWSE_ABANDONMENT_COLLECTION_PAGE_TYPE,
+        self::BROWSE_ABANDONMENT_PRODUCT_PAGE_TYPE
+    ];
+
+    /**
+     * @var YotpoConfig
+     */
+    private $yotpoConfig;
+
+    /**
+     * @var Registry
+     */
+    private $coreRegistry;
+
+    /**
+     * @var CustomerModelSession
+     */
+    private $customerModelSession;
+
+    /**
+     * @var CustomerModelSession
+     */
+    private $request;
+
+    /**
+     * @var CheckoutSessionFactory
+     */
+    private $checkoutSessionFactory;
+
+    /**
+     * BrowseAbandonment constructor.
+     * @param Context $context
+     * @param YotpoConfig $yotpoConfig
+     * @param Registry $coreRegistry
+     * @param CustomerModelSession $customerModelSession
+     * @param HttpRequest $request
+     * @param CheckoutSessionFactory $checkoutSessionFactory
+     * @param array<mixed> $data
+     */
+    public function __construct(
+        Context $context,
+        YotpoConfig $yotpoConfig,
+        Registry $coreRegistry,
+        CustomerModelSession $customerModelSession,
+        HttpRequest $request,
+        CheckoutSessionFactory $checkoutSessionFactory,
+        array $data = []
+    ) {
+        $this->yotpoConfig = $yotpoConfig;
+        $this->coreRegistry = $coreRegistry;
+        $this->customerModelSession = $customerModelSession;
+        $this->request = $request;
+        $this->checkoutSessionFactory = $checkoutSessionFactory;
+        parent::__construct($context, $data);
+    }
+
+    /**
+     * Check module is enabled and API tokens set
+     *
+     * @return boolean
+     * @throws LocalizedException
+     * @throws NoSuchEntityException
+     */
+    public function isAppKeyAndSecretSet()
```

Comment:
let's change this from `isAppKeyAndSecretSet` -> `isYotpoExtensionActive` or somthing similar, WDYT?

Reply:
But it doesn't check that the extension is active, it really checks only if app key and secret are properly set :)

Reply:
So if the extension is not active will it still process browse abandonment events? just making sure

Reply:
It will not, the block will not be triggered as per the `ifconfig` in `default.xml`.